drivers/flash/flash_shell.c

```diff
@@ -255,6 +255,158 @@ static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
 	return result;
 }
 
+static int set_bypass(const struct shell *shell, shell_bypass_cb_t bypass)
+{
+	static bool in_use;
+
+	if (bypass && in_use) {
+		shell_error(shell, "flash load supports setting bypass on a single instance.");
+
+		return -EBUSY;
+	}
+
+	/* Mark that we have set or unset the bypass function */
+	in_use = bypass != NULL;
+
+	if (in_use) {
+		shell_print(shell, "Loading...");
+	}
+
+	shell_set_bypass(shell, bypass);
+
+	return 0;
+}
+
+#define FLASH_LOAD_BUF_MAX 256
+
+static const struct device *flash_load_dev;
+static uint32_t flash_load_addr;
+static uint32_t flash_load_total;
+static uint32_t flash_load_written;
+static uint32_t flash_load_chunk;
+
+static uint32_t flash_load_boff;
+static uint8_t flash_load_buf[FLASH_LOAD_BUF_MAX];
+
+static void bypass_cb(const struct shell *shell, uint8_t *recv, size_t len)
+{
+	uint32_t left_to_read = flash_load_total - flash_load_written - flash_load_boff;
+	uint32_t to_copy = MIN(len, left_to_read);
+	uint32_t copied = 0;
+
+	while (copied < to_copy) {
+
+		uint32_t buf_copy = MIN(to_copy, FLASH_LOAD_BUF_MAX - flash_load_boff);
+
+		memcpy(flash_load_buf + flash_load_boff, recv + copied, buf_copy);
+
+		flash_load_boff += buf_copy;
+		copied += buf_copy;
+
+		/* Buffer is full. Write data to memory. */
+		if (flash_load_boff == FLASH_LOAD_BUF_MAX) {
+			uint32_t addr = flash_load_addr + flash_load_written;
+			int rc = flash_write(flash_load_dev, addr, flash_load_buf,
+					FLASH_LOAD_BUF_MAX);
+
+			if (rc != 0) {
+				shell_error(shell, "Write to addr %x on dev %p ERROR!",
+						addr, flash_load_dev);
+			}
+
+			shell_print(shell, "Written chunk %d", flash_load_chunk);
+
+			flash_load_written += FLASH_LOAD_BUF_MAX;
+			flash_load_chunk++;
+			flash_load_boff = 0;
+		}
+	}
+
+	/* When data is not aligned to FLASH_LOAD_BUF_MAX there may be partial write
+	 * at the end.
+	 */
+	if (flash_load_written < flash_load_total &&
+			flash_load_written + flash_load_boff >= flash_load_total) {
+
+		uint32_t addr = flash_load_addr + flash_load_written;
+		int rc = flash_write(flash_load_dev, addr, flash_load_buf, flash_load_boff);
+
+		if (rc != 0) {
+			set_bypass(shell, NULL);
+			shell_error(shell, "Write to addr %x on dev %p ERROR!",
+					addr, flash_load_dev);
+			return;
+		}
+
+		shell_print(shell, "Written chunk %d", flash_load_chunk);
+		flash_load_written += flash_load_boff;
+		flash_load_chunk++;
+	}
+
+	if (flash_load_written >= flash_load_total) {
+		set_bypass(shell, NULL);
+		shell_print(shell, "Read all");
+	}
+}
+
+static int cmd_load(const struct shell *shell, size_t argc, char *argv[])
```

Comment:
Same here. Currently, it is not allowed for the variable and structure name to be the same.